ddcommon-ffi/build.rs

```diff
@@ -0,0 +1,16 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+extern crate build_common;
+
+use build_common::generate_header;
+use std::env;
+use std::path::PathBuf;
+
+fn main() {
+    let crate_dir = PathBuf::from(env::var_os("CARGO_MANIFEST_DIR").unwrap());
+    let header_name = "common.h";
+    let output_base_dir = env::var("DESTDIR").ok(); // Use `ok()` to convert Result to Option
+
+    generate_header(crate_dir, header_name, output_base_dir.as_deref());
+    println!("cargo:rerun-if-env-changed=DESTDIR");
```

Comment:
We have the exact same code for all three crates, with the only difference being the header name.  Maybe worth factoring into a function?

Reply:
I have mixed feelings about hiding some of this behaviour :-)
I added the magical function all the same.